app/src/main/java/com/example/tasky/login/components/TaskyTextField.kt

```diff
@@ -0,0 +1,145 @@
+import com.example.tasky.ui.theme.LightBlue2
+import com.example.tasky.ui.theme.LightGray2
+import com.example.tasky.ui.theme.Red
+
+@Composable
+fun TaskyTextField(
+    modifier: Modifier = Modifier,
+    value: TextFieldValue,
+    isValidEmail: Boolean = false,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Done,
+    onValueChange: (TextFieldValue) -> Unit,
+) {
+    BaseTaskyTextField(
+        modifier = modifier,
+        value = value,
+        onValueChange = onValueChange,
+        isValidEmail = isValidEmail,
+        keyboardType = keyboardType,
+        imeAction = imeAction
+    )
+}
+
+@Composable
+fun PasswordTaskyTextField(
+    modifier: Modifier = Modifier,
+    value: TextFieldValue,
+    showPassword: Boolean = false,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Done,
+    onShowPasswordChange: (Boolean) -> Unit = {},
+    onValueChange: (TextFieldValue) -> Unit,
+) {
+    BaseTaskyTextField(
+        modifier = modifier,
+        value = value,
+        isPassword = true,
+        showPassword = showPassword,
+        onValueChange = onValueChange,
+        keyboardType = keyboardType,
+        imeAction = imeAction,
+        onShowPasswordChange = onShowPasswordChange
+    )
+}
+
+@Composable
+private fun BaseTaskyTextField(
+    modifier: Modifier = Modifier,
+    value: TextFieldValue,
+    isPassword: Boolean = false,
+    isValidEmail: Boolean = false,
+    showPassword: Boolean = false,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Done,
+    onValueChange: (TextFieldValue) -> Unit,
+    onShowPasswordChange: (Boolean) -> Unit = {}
+) {
+    OutlinedTextField(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        value = value,
+        onValueChange = {
+            onValueChange(value)
+        },
+        keyboardOptions = KeyboardOptions(
+            keyboardType = keyboardType,
+            imeAction = imeAction
+        ),
+        textStyle = MaterialTheme.typography.bodyMedium,
+        shape = RoundedCornerShape(10.dp),
+        colors = TextFieldDefaults.colors(
+            errorCursorColor = Red,
+            errorIndicatorColor = Red,
+            errorLabelColor = Red,
+            errorContainerColor = LightGray2,
+            errorTrailingIconColor = Gray2,
+            focusedTextColor = DarkGray,
+            focusedPlaceholderColor = Gray,
+            focusedLeadingIconColor = Gray,
+            focusedContainerColor = LightGray2,
+            focusedIndicatorColor = LightBlue2,
+            focusedTrailingIconColor = if (isValidEmail) Green else Gray2,
+            unfocusedContainerColor = LightGray2,
+            unfocusedIndicatorColor = Color.Transparent,
+            unfocusedTrailingIconColor = if (isValidEmail) Green else Gray2,
+            cursorColor = DarkGray
+        ),
+        trailingIcon = {
```

Comment:
To make this a bit easier to read, you can create a small function
"getTrailingIcon(params...)"